web/modules/custom/usergreeting/src/GreetingTime.php

```diff
@@ -61,15 +72,18 @@ public function __construct(TimeInterface $timeOutput, AccountInterface $account
    */
   public function greetingMessage(): array {
     $time_output = (int) date('G', $this->timeOutput->getRequestTime());
+    $config_morning = $this->configFactory->get('usergreeting.settings')->get('morning_start');
+    $config_afternoon = $this->configFactory->get('usergreeting.settings')->get('afernoon_start');
+    $config_evening = $this->configFactory->get('usergreeting.settings')->get('evening_start');
 
-    if ($time_output >= 5 && $time_output < 12) {
-      $greeting = $this->t('Good Morning');
+    if ($time_output >= $config_morning && $time_output < $config_afternoon) {
+      $greeting = implode(' ', [$greeting = $this->t('Good Morning')]);
```

Comment:
Unfortunately implode doesn't do anything when you use it like this. Firstly, you don't need to set the value twice, it would be enough to pass `[$this->t('Message')]` into the array. However, implode glues together pieceS of an array into a single string, so if you pass an array with a single element it cannot glue that element to itself, it's only useful if the array contains multiple elements.

If you want the final message to look something like "Good morning User!" then you need to do glue together:

1. the message
1. the username
1. the exclamation mark

But do you really need implode? See my other comment.